scripts/mbedtls_framework/test_case.py

```diff
@@ -89,3 +91,33 @@ def write_data_file(filename: str,
             tc.write(out)
         out.write('\n# End of automatically generated file.\n')
     os.replace(tempfile, filename)
+
+def hash_dependency_symbol(psa_hash_alg: str) -> str:
```

Comment:
I think the function name should include something like `compat36`. Also, the description should lead with the fact that it's about 3.6 compatibility: for us it's obvious due the context of the PR, but otherwise "legacy or USE_PSA_CRYPTO domain" in the first sentence will make little sense to future readers as those only exist in 3.6.